### DISCUSSION TITLE
Support saving JSON output to the filesystem

### DISCUSSION THREAD

**Discussion by cfjedimaster** [11ty/eleventy#4042] [Enhancement Queue]
**Is your feature request related to a problem? Please describe.**
The new `--to` argument is pretty cool, but I noticed that when I did this:

```
eleventy --to=json > foo.json
```

That any console message from my code was included in the output. That's to be expected as I piped everything to a file and I could fix that by removing any console logging, but what if we could have the output stored to the filesystem automatically? 

**Describe the solution you'd like**
What if the argument to --to allowed filenames. So `to=json` works as expected, but `to=foo.json` saves the JSON to `./foo.json`. 
Ditto for ndjson as well. This would allow for 5 possibilities:

* fs (default)
* json - output JSON
* ndjson - ditto
* something.json - save to something.json
* something.ndjson - save to something.ndjson

**Comment by zachleat** [comment]
I like this!

Programmatic Eleventy would be a good short-term solution here too.

```
let elev = new Eleventy("./test/stubs--to/");
await elev.init();
let result = await elev.toJSON();
// write result to file
```

https://github.com/11ty/eleventy/blob/861d5c9b909b7104b1619c48f7e59501da1bc6ce/test/EleventyTest.js#L167

**Comment by zachleat** [comment]
> This repository is now using [lodash style issue management](https://twitter.com/samselikoff/status/991395669016436736) for enhancements. This means enhancement issues will now be closed instead of leaving them open.

> View [the enhancement backlog here](https://github.com/11ty/eleventy/issues?utf8=%E2%9C%93&q=label%3Aneeds-votes+sort%3Areactions-%2B1-desc+). Don’t forget to upvote the top comment with 👍!
